### PR TITLE
Add support for partial foot contact in the planner and controller

### PR DESCRIPTION
**Pull request by rdeits**
This adds a new walking parameter, `support_contact_groups` which by default is `{'heel', 'toe'}` but can be set to, for example, `{'midfoot', 'toe'}` to only use the front half of the foot for support, both in the planner and in the controller.

### CODE REVIEW

**Comment by RussTedrake** [comment]
@avalenzu is looking at the testTaskSpaceRRT failure.

File: systems/robotInterfaces/@Biped/planZMPTraj.m [hunk 1]
```diff
@@ -43,20 +43,21 @@
...
-  z = steps.right(1).pos;
-  zmp0(:,end+1) = z(1:2);
-  initial_supports(end+1) = biped.foot_body_id.right;
-end
-if steps.left(1).is_in_contact
-  z = steps.left(1).pos;
-  zmp0(:,end+1) = z(1:2);
-  initial_supports(end+1) = biped.foot_body_id.left;
+com0 = getCOM(biped, kinsol);
+sole_to_com = rotmat(-footsteps(1).pos(6)) * (com0(1:2) - footsteps(1).pos(1:2));
```
Comment (RussTedrake) [inline systems/robotInterfaces/@Biped/planZMPTraj.m]: is pos(6) correct here?

**Comment by RussTedrake** [comment]
code looks good to me.

**Comment by rdeits** [comment]
Yeah, the footsteps are still using RPY notation for now.

**Comment by RussTedrake** [comment]
looks like the errors in systems/plants/constraint/test/testKinCnst (and systems/plants/test/testIK ) are also real...?

**Comment by rdeits** [comment]
testKinCnst was indeed assuming that each foot had exactly 4 contact points. I've resolved that one.

**Comment by rdeits** [comment]
OK, I've fixed testIK and testKinCnst

**Pull request merged**